src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -125,7 +126,10 @@ class InstanceOpFactoryImpl(
         config.defaultAcceptedResourceRolesSet, config, schedulerPlugins, localRegion)
     matchResponse match {
       case matches: ResourceMatchResponse.Match =>
-        val taskId = Task.Id.forInstanceId(scheduledInstance.instanceId)
+
+        val taskId = scheduledInstance.tasksMap.values.headOption.map(_.taskId).map(Task.Id.forResidentTask).getOrElse {
```

Comment:
No I'm not re-using taskIds. `Task.Id.forResidentTask` will create a taskId with an incremented attempt count for an existing taskId; that's how we do it for resident tasks.

This should probably be renamed to something like `withIncarnationCount` or whatever. I didn't do much renaming since that would have increased the loc to review, but I'm happy to rename if we're good with the overall change.